src/main/cpp/filewatchdog.cpp

```diff
@@ -74,11 +74,7 @@ void FileWatchdog::stop()
 {
 	LogLog::debug(LOG4CXX_STR("Stopping file watchdog"));
 	m_priv->interrupted = 0xFFFF;
-
-	{
-		std::unique_lock<std::mutex> lock(m_priv->interrupt_mutex);
-		m_priv->interrupt.notify_all();
-	}
+	m_priv->interrupt.notify_all();
```

Comment:
There must be a mutex lock between the condition change and cv-notify. Without such a lock the notification can be loosed. Please, take an attention. An atomic condition cannot be a reason for cancellation of a mutex lock, cv logic absolutely requires such a lock.

Reply:
Thank you for noticing this problem. I have opened #371 to address this.